packages/entities/entities-redis-configurations/src/components/RedisConfigurationSelector.vue

```diff
@@ -0,0 +1,205 @@
+        data-testid="new-redis-config-area"
+        @click="onCreateNew"
+      >
+        <AddIcon :size="KUI_ICON_SIZE_20" />
+        <span>{{ createButtonText }}</span>
+      </div>
+    </template>
+  </KSelect>
+
+  <!-- New Redis Configuration Modal -->
+  <RedisConfigurationFormModal
+    :partial-type="redisType"
+    :visible="isModalVisible"
+    @created="onPartialCreated"
+    @modal-close="onModalClose"
+  />
+</template>
+
+<script setup lang="ts">
+import { computed, watch, ref } from 'vue'
+import { AddIcon } from '@kong/icons'
+import { KUI_ICON_SIZE_20 } from '@kong/design-tokens'
+import type { SelectItem } from '@kong/kongponents'
+import { useRedisConfigurationSelector } from '../composables/useRedisConfigurationSelector'
+import useI18n from '../composables/useI18n'
+import RedisConfigurationFormModal from './RedisConfigurationFormModal.vue'
+import type { RedisConfigurationResponse } from '../types'
+
+const {
+  redisType = 'redis-ee',
+  showCreateButton = true,
+  ...props
+} = defineProps<{
+  /** The selected redis configuration ID */
+  modelValue?: string
+  /** Redis type filter: 'redis-ce', 'redis-ee' */
+  redisType?: 'redis-ce' | 'redis-ee'
+  /** Placeholder text for the selector */
+  placeholder?: string
+  /** Text to show when no items are available */
+  emptyStateText?: string
+  /** Whether to show the create new configuration button */
+  showCreateButton?: boolean
+  /** Text for the create new configuration button */
+  createButtonText?: string
+}>()
+
+const emit = defineEmits<{
+  /** Emitted when the selection changes */
+  'update:modelValue': [value: string | undefined]
+  /** Emitted when a configuration is selected */
+  'change': [item: SelectItem | null]
+  /** Emitted when the create new button is clicked */
+  'create-new': []
+  /** Emitted when an error occurs */
+  'error': [error: Error]
+  'modal-close': []
+}>()
+
+const { i18n: { t } } = useI18n()
+
+const {
+  items,
+  loading,
+  onQueryChange,
+  error,
+  loadItems,
+} = useRedisConfigurationSelector({
+  redisType,
+})
+
+// Modal state
+const isModalVisible = ref(false)
+
+// Computed properties for default text values
+const placeholder = computed(() =>
+  props.placeholder || t('selector.placeholder'),
+)
+
+const emptyStateText = computed(() =>
+  props.emptyStateText || t('selector.empty_state'),
+)
+
+const createButtonText = computed(() =>
+  props.createButtonText || t('selector.create_new'),
+)
+
+// Event handlers
+const onSelectionChange = (item: SelectItem<string> | null) => {
+  const value = item?.value
+  emit('update:modelValue', value as string)
+  emit('change', item)
+}
+
+const onCreateNew = () => {
+  isModalVisible.value = true
+}
+
+// Modal event handlers
+const onModalClose = () => {
+  isModalVisible.value = false
+  emit('modal-close')
+}
+
+const onPartialCreated = (data: RedisConfigurationResponse) => {
+  isModalVisible.value = false
+  loadItems() // Refresh the list
```

Comment:
Should this be awaited?

Reply:
Not necessary, `onSelectionChange` doesn't relay on the refreshing